File: app/src/main/java/com/github/kongpf8848/animation/activity/tween/scale/WeizhiActivity.java
package com.github.kongpf8848.animation.activity.tween.scale;

import android.animation.AnimatorSet;
import android.animation.ObjectAnimator;
import android.animation.PropertyValuesHolder;
import android.os.Bundle;
import android.os.Looper;
import android.os.MessageQueue;
import android.widget.ImageView;

import com.github.kongpf8848.animation.R;
import com.github.kongpf8848.animation.activity.BaseActivity;

import java.util.Random;

import butterknife.BindView;


/**
 * Created by jack on 2016/8/6.
 */
public class WeizhiActivity extends BaseActivity {

    @BindView(R.id.iv_activity_splash_img)
    public ImageView mIvActivitySplashImg;

    @BindView(R.id.iv_activity_splash_pic)
    public ImageView mIvActivitySplashPic;

    @Override
    protected int getLayoutId() {
        return R.layout.activity_weizhi;
    }

    @Override
    protected void onCreateEnd(Bundle savedInstanceState){
        super.onCreateEnd(savedInstanceState);
        int index = 1 + new Random().nextInt(3);
        int resourceId = getResources().getIdentifier("welcome_icon_" + index, "mipmap", getPackageName());
        this.mIvActivitySplashPic.setImageResource(resourceId);
        this.mIvActivitySplashPic.setAlpha(0.0F);

        Looper.myQueue().addIdleHandler(new MessageQueue.IdleHandler() {
            @Override
            public boolean queueIdle() {
                startAnim();
                return false;
            }
        });
    }

    @Override
    public void onDestroy() {
        super.onDestroy();
        this.mIvActivitySplashImg.clearAnimation();
        this.mIvActivitySplashPic.clearAnimation();
    }


    private void startAnim() {
        ObjectAnimator animator1 = ObjectAnimator.ofFloat(this.mIvActivitySplashImg, "alpha", 1.0F, 0.0F);
        animator1.setDuration(1000L);
        ObjectAnimator animator2 = ObjectAnimator.ofFloat(this.mIvActivitySplashPic, "alpha", 0.0F, 1.0F);
        animator2.setDuration(1000L);

        PropertyValuesHolder valuesHolder1 = PropertyValuesHolder.ofFloat("scaleX", 1.0f, 1.1f);
        PropertyValuesHolder valuesHolder2 = PropertyValuesHolder.ofFloat("scaleY", 1.0f, 1.1f);
        ObjectAnimator animator3 = ObjectAnimator.ofPropertyValuesHolder(this.mIvActivitySplashPic, valuesHolder1, valuesHolder2);
        animator3.setDuration(1000L);

        AnimatorSet set = new AnimatorSet();
        set.play(animator1).with(animator2).before(animator3);
        set.start();
    }


}

File: app/src/main/java/com/github/kongpf8848/animation/activity/tween/scale/WeizhiActivity.kt
package com.github.kongpf8848.animation.activity.tween.scale

import android.animation.AnimatorSet
import android.animation.ObjectAnimator
import android.animation.PropertyValuesHolder
import android.os.Bundle
import android.os.Looper
import android.view.View
import com.github.kongpf8848.animation.R
import com.github.kongpf8848.animation.activity.BaseActivity
import kotlinx.android.synthetic.main.activity_weizhi.*
import java.util.*

class WeizhiActivity : BaseActivity() {

    override fun getLayoutId(): Int {
        return R.layout.activity_weizhi
    }

    override fun fitsSystemWindows(): Boolean {
        return false
    }

    override fun statusBarColor(): Int {
        return R.color.transparent
    }

    override fun onCreateEnd(savedInstanceState: Bundle?) {
        super.onCreateEnd(savedInstanceState)
        val index = 1 + Random().nextInt(3)
        val resourceId = resources.getIdentifier("welcome_icon_$index", "mipmap", packageName)
        iv_activity_splash_pic.setImageResource(resourceId)
        iv_activity_splash_pic.alpha = 0.0f

        Looper.myQueue().addIdleHandler {
            startAnimation()
            false
        }
    }

    public override fun onDestroy() {
        super.onDestroy()
        iv_activity_splash_bg.clearAnimation()
        iv_activity_splash_pic.clearAnimation()
    }

    private fun startAnimation() {
        val animator1 = ObjectAnimator.ofFloat(iv_activity_splash_bg, View.ALPHA, 1.0f, 0.0f)
        val animator2 = ObjectAnimator.ofFloat(iv_activity_splash_pic, View.ALPHA, 0.0f, 1.0f)
        val animator3 = ObjectAnimator.ofPropertyValuesHolder(iv_activity_splash_pic, PropertyValuesHolder.ofFloat(View.SCALE_X, 1.0f, 1.1f), PropertyValuesHolder.ofFloat(View.SCALE_Y, 1.0f, 1.1f))
        AnimatorSet().setDuration(1000L).apply {
            play(animator1).with(animator2).before(animator3)
            start()
        }
    }
}